### PR TITLE
Remove `warningMessages`

### PR DESCRIPTION
**Pull request by basil**
As a first step toward #407, get rid of `warningMessages`. These were passed around from one class to another and ultimately logged in an XML file that nobody ever looks at. In the handful of cases where we were actually trying to log a warning, I changed the warning to a log statement so that it now gets printed to the console where someone can actually see it. To test this I ran `jenkinsci/bom` tests with `ace-editor` (saw the warning) and `text-finder` (saw no warning), which seems sufficient for this low-risk change.

### CODE REVIEW

File: src/main/java/org/jenkins/tools/test/model/PomData.java [hunk 1]
```diff
@@ -77,12 +72,4 @@ public String getPackaging() {
...
         return scmTag;
     }
-
-    public boolean isPluginPOM() {
-        if (parent != null) {
-            return parent.matches("org.jenkins-ci.plugins", "plugin");
-        } else { // Interpolate by packaging
-            return "hpi".equalsIgnoreCase(packaging);
-        }
-    }
```
Comment (basil) [inline src/main/java/org/jenkins/tools/test/model/PomData.java:87]: Unrelated cleanup while I was here: removing this now unused code.

File: src/main/java/org/jenkins/tools/test/model/MavenCoordinates.java [hunk 2]
```diff
@@ -83,15 +83,6 @@ public String toString(){
...
 
-    public String toGAV(){
-        return groupId+":"+artifactId+":"+version;
-    }
-
-    public static MavenCoordinates fromGAV(String gav){
-        String[] chunks = gav.split(":");
-        return new MavenCoordinates(chunks[0], chunks[1], chunks[2]);
-    }
-
```
Comment (basil) [inline src/main/java/org/jenkins/tools/test/model/MavenCoordinates.java:94]: Unrelated cleanup: removing these unused methods while I was here.

**Review by imonteroperez — APPROVED** [review]

**Review by jtnord — APPROVED** [review]
